src/javascript/app_2/Modules/Trading/Components/Form/Purchase/contract_info.jsx

```diff
@@ -3,60 +3,63 @@ import React             from 'react';
 import { localize }      from '_common/localize';
 import Money             from 'App/Components/Elements/money.jsx';
 import Tooltip           from 'App/Components/Elements/tooltip.jsx';
-import { IconTradeType } from 'Assets/Trading/Types';
+import { IconPriceMove } from 'Assets/Trading/icon_price_move.jsx';
 
-const ContractInfo = ({
-    barrier_count,
-    contract_title,
-    contract_type,
-    currency,
-    proposal_info,
-}) => {
-    const icon_type = `${contract_type}${/^(call|put)$/i.test(contract_type) && barrier_count > 0 ? '_barrier' : ''}`.toLowerCase();
+class ContractInfo extends React.PureComponent {
+    constructor(props) {
+        super(props);
 
-    return (
-        <div className='box'>
-            <div className='left-column'>
-                <div className='type-wrapper'>
-                    <IconTradeType type={icon_type} className='type' />
-                </div>
-                <h4 className='trade-type'>{contract_title}</h4>
-            </div>
-            {(proposal_info.has_error || !proposal_info.id) ?
-                <div className={proposal_info.has_error ? 'error-info-wrapper' : ''}>
-                    <span>{proposal_info.message}</span>
-                </div>
-                :
-                <div className='purchase-info-wrapper'>
-                    <span className='purchase-tooltip'>
-                        <Tooltip alignment='left' icon='info' message={proposal_info.message} />
-                    </span>
-                    <div className='info-wrapper'>
-                        <div>{localize('Stake')}:</div>
-                        <div><Money amount={proposal_info.stake} currency={currency} /></div>
-                    </div>
-                    <div className='info-wrapper'>
-                        <div>{localize('Payout')}:</div>
-                        <div><Money amount={proposal_info.payout} currency={currency} /></div>
-                    </div>
-                    <div className='info-wrapper'>
-                        <div>{localize('Net Profit')}:</div>
-                        <div><Money amount={proposal_info.profit} currency={currency} /></div>
+        this.previous_price = null;
+    }
+
+    getIconType = () => {
+        let icon_type = '';
+        const contract_basis = this.props.contract_basis.value;
+        const price = this.props.proposal_info[contract_basis];
+        if (price !== this.previous_price) {
+            icon_type = price > this.previous_price ? 'profit' : 'loss';
+            this.previous_price = price;
```

Comment:
i think it's better to use `state` and `setState` for `previous_price` and `icon_type`. you can define it before constructor as such: 
```
state = {
   previous_price: null,
   icon_type: null,
};
```